packages/bodiless-components/src/RedirectAlias/form.tsx

```
@@ -0,0 +1,248 @@
+  HOC,
+  addClasses,
+  removeClasses,
+} from '@bodiless/fclasses';
+import type {
+  ContextMenuFormProps,
+} from '@bodiless/core';
+import { ComponentFormDefaultPanelSize } from '@bodiless/ui';
+import { withToolsButton } from '../Tools';
+import { useGetRedirectAliases } from './hooks';
+import type { AliasItem } from './types';
+
+enum Steps { Edit, Confirmation }
+
+const REDIRECT_ALIASES = 'Redirect Aliases';
+const CONFIRMATION = 'Redirect aliases file validated and saved.';
+const INVALIDATED = 'The redirects are not valid, please correct.';
+const DEFAULT_REDIRECT_STATUS = '301';
+const ALIASPARTSCOUNT = 3;
+
+const isTextEmpty = (text: string) => (!text || text === '');
+
+const isTextValidate = (text: string): boolean => {
+  // Users must be able to save no redirects.
+  if (isTextEmpty(text)) return true;
+
+  // Any double whitespaces are invalid.
+  if (text.indexOf('  ') >= 0) return false;
+
+  try {
+    const aliases = text.split('\n');
+    const validatedAliases = aliases.filter(item => {
+      // For each item, first and last chars must not be whitespace.
+      if (item[0] == ' ' || item[item.length - 1] == ' ') {
+        return false;
+      }
+
+      // Now, split the item values so we can validate each one individually.
+      const items = item.split(' ');
+
+      // Items are valid if last value is not provided.
+      // First and second values can not be numbers.
+      // Last, if provided, must be validated as a number to represent status code.
+      if (
+        items.length !== ALIASPARTSCOUNT && items.length !== (ALIASPARTSCOUNT - 1)
+        || typeof items[0] !== 'string' || !isNaN(parseInt(items[0]))
+        || typeof items[1] !== 'string' || !isNaN(parseInt(items[1])) 
+        || typeof items[2] !== 'undefined' && isNaN(parseInt(items[2]))
+      ) {
+        return false;
+      }
+
+      return true;
+    });
+
+    return validatedAliases.length === aliases.length;
+  } catch (error) {
+    return false;
```

Could this cause silent errors? The only possible error I can foresee happening here is accessing an out-of-bounds array index, but maybe we should at least log the error in the console, just in case. Feel free to ignore this one, though.

---

Hello @lucas-varela ,

I don't think so since it will be triggered anyways as invalid message for the admin user because, probably in this case, the expected pattern will not be followed (which can cause the example you mentioned about out-of-bounds). However, I think it's not harmful to put a console error there so I updated it.